content/java/concepts/stack/stack.md

```diff
@@ -0,0 +1,87 @@
+---
+Title: 'Stack'
+Description: 'A Stack is a concrete class that offers operations for accessing or manipulating items at the top of a stack, a linear data structure that follows the last-in, first-out (LIFO) rule.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Collections'
+  - 'Stack'
+  - 'Data Structures'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+A `Stack` is a concrete class provided in the `java.util` package that extends the legacy `Vector` class. It is a linear data structure which supports retrieval by last-in, ﬁrst-out (LIFO) order. It means that the insert and remove operations only occur on the top of the stack. The insert/put operation is called `push`, whereas the remove is called `pop`.
+
+An opposite data structure is the [Queue](https://www.codecademy.com/resources/docs/java/queue) interface, which follows the first-in, first-out (FIFO) order. Both `Stack` and `Queue` concepts are present in our daily life (e.g., when people are standing in a queue to buy entry ticket for watching a soccer game or when we have a stack of dishes waiting to be washed!).
+
+## Syntax
+
+```java
+import java.util.Stack;
+
+Stack<DataType> s = new Stack<>();
+```
+Where `DataType` is the data type to be stored in the stack. Note that, differently from `Queue`, Stack is a concrete class, not an interface.
+
+## Methods
+
+The `Stack` class provides the following methods:
+
+- `.push()` : `.push(item)`  adds  `item`  onto the top of the  `Stack`  .
+- `.pop()` : removes and returns the object at the top of the  `Stack`, throwing an exception when the `Stack` is empty.
+- `.peek()` : returns the head (top) of the `Stack` without removing it. It throws an exception when the `Stack` is empty.
+- `.empty()` : returns `true` if the `Stack` contains no items. Otherwise, it returns `false`.
+- `.search(item)` : returns the distance of `item` from the top of the stack, starting from `1`. A `-1` value is returned if `item` is not on the stack.
```

Comment:
```suggestion
- `.push(item)`: adds an item to the top of the `Stack`.
- `.pop()`: removes and returns the object at the top of the `Stack`, throwing an exception when the `Stack` is empty.
- `.peek()`: returns the (top) of the `Stack` without removing it, and throws an exception when the `Stack` is empty.
- `.empty()`: returns `true` if the `Stack` contains no items or `false`, otherwise.
- `.search(item)`: returns the distance the `item` is from the top of the `Stack`, starting from `1`, or `-1` is if `item` is not in the `Stack`.
```